diff --git a/src/webp/types.h b/src/webp/types.h
index 0ce2622..b810e1b 100644
--- a/src/webp/types.h
+++ b/src/webp/types.h
@@ -37,6 +37,16 @@ typedef long long int int64_t;
 #endif  /* _MSC_VER */

 #ifndef WEBP_EXTERN
+  #ifdef   _MSC_VER
+    #ifdef WEBP_DLL
+        #define WEBP_EXTERN __declspec(dllexport)
+    #else
+        #define WEBP_EXTERN __declspec(dllimport)
+    #endif
+  #endif /* _MSC_VER */
+#endif
+
+#ifndef WEBP_EXTERN
 // This explicitly marks library functions and allows for changing the
 // signature for e.g., Windows DLL builds.
 # if defined(__GNUC__) && __GNUC__ >= 4
